### PR TITLE
Service discovery feature

### PR DESCRIPTION
**Pull request by wiseflat**
Hi,

I am using coredns as a secondary dns resolver.
Main goal: adding service discovery to haproxy

[Link](https://www.haproxy.com/documentation/aloha/9-5/traffic-management/lb-layer7/dns-srv-records/)

```
resolvers test
   nameserver dns1 192.168.0.2:5353

backend be_app
   balance            roundrobin
   cookie             SERVERID insert indirect nocache
   server-template srv 3 _http._tcp.web.skydns.local resolvers test check
```

### CODE REVIEW

**Comment by tersmitten** [comment]
Thanks for your contribution. I'll try to test and merge this soon.

**Comment by tersmitten** [comment]
* https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#5.3.2-resolvers
* https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#5.3.2-nameserver
* https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#4.2-server-template

File: README.md [hunk 1]
```diff
@@ -68,6 +68,11 @@ Set up (the latest version of) [HAProxy](http://www.haproxy.org/) in Ubuntu syst
 * `haproxy_defaults_compression`: [optional]: Compression declarations
 * `haproxy_defaults_compression.{}.name`: [required]: The compression name (e.g. `algo`, `type`, `offload`)
 * `haproxy_defaults_compression.{}.value`: [required]: The compression value, (e.g. if name = algo : one of this values `identity`, `gzip`, `deflate`, `raw-deflate` / if name = type : list of mime type separated by space for example `text/html text/plain text/css` / if name = `offload` value is empty)
+  
+* `haproxy_resolvers_label`: [default: `[]`]: The global name of your dns resolver
+* `haproxy_resolvers_list.{n}.name`: [required]: The name of the dns resolver
```
Comment (tersmitten) [inline README.md]: Can you rename this to `id`

File: README.md [hunk 2]
```diff
@@ -68,6 +68,11 @@ Set up (the latest version of) [HAProxy](http://www.haproxy.org/) in Ubuntu syst
 * `haproxy_defaults_compression`: [optional]: Compression declarations
 * `haproxy_defaults_compression.{}.name`: [required]: The compression name (e.g. `algo`, `type`, `offload`)
 * `haproxy_defaults_compression.{}.value`: [required]: The compression value, (e.g. if name = algo : one of this values `identity`, `gzip`, `deflate`, `raw-deflate` / if name = type : list of mime type separated by space for example `text/html text/plain text/css` / if name = `offload` value is empty)
+  
+* `haproxy_resolvers_label`: [default: `[]`]: The global name of your dns resolver
```
Comment (tersmitten) [inline README.md]: Can you remove this variable and make `haproxy_resolvers_list` look like `haproxy_userlists`?

File: templates/etc/haproxy/haproxy.cfg.j2 [hunk 3]
```diff
@@ -6,6 +6,8 @@ global
 defaults
 {% include 'defaults.cfg.j2' %}
 
+{% include 'resolvers.cfg.j2' %}
```
Comment (tersmitten) [inline templates/etc/haproxy/haproxy.cfg.j2]: `resolverslist`

File: README.md (hunk 2 above)
Comment (tersmitten) [reply README.md]: `haproxy_resolverslist`

**Comment by wiseflat** [comment]
Hi @tersmitten  👍 

I will submit a new merge request soon but you will have to confirm me what you wish. I am not sure about some variables.

**Comment by tersmitten** [comment]
Thanks

File: templates/etc/haproxy/backend.cfg.j2 [hunk 4]
```diff
@@ -106,10 +106,19 @@ backend {{ backend.name }}
   compression {{ compression.name }} {{ compression.value }}
 {% endfor %}
 
+{% if backend.server is defined %}
```
[the remark this answers is not in the document]
Comment (wiseflat) [reply templates/etc/haproxy/backend.cfg.j2]: You have to choose between server and server_template. Maybe you could have both?
AFAIK, If backend.server is not defined, ninja templating will failed if you don't check this.

File: README.md (hunk 1 above)
Comment (wiseflat) [reply README.md]: Please @tersmitten, give me the line and the detail of your need, I am really not sure about what you wish

File: README.md [hunk 5]
```diff
@@ -139,6 +144,12 @@ Set up (the latest version of) [HAProxy](http://www.haproxy.org/) in Ubuntu syst
 * `haproxy_listen.{n}.server.{n}.name`: [required]: The internal name assigned to this server
 * `haproxy_listen.{n}.server.{n}.listen`: [required]: Defines a listening address and/or ports
 * `haproxy_listen.{n}.server.{n}.param`: [optional]: A list of parameters for this server
+* `haproxy_listen.{n}.server_template`: [optional]: Server template declarations
+* `haproxy_listen.{n}.server_template.{n}.name`: [required]: The internal name assigned to this server
```
[the remark this answers is not in the document]
Comment (wiseflat) [reply README.md]: Please @tersmitten, give me the line and the detail of your need, I am really not sure about what you wish

**Comment by tersmitten** [comment]
See https://github.com/Oefenweb/ansible-haproxy/releases/tag/v6.7.0 (and later). Should contain anything you need

**Comment by tersmitten** [comment]
Feel free to open a new PR if not

**Pull request closed without merging**